colossalai/nn/optimizer/cpu_adam.py

```diff
@@ -1,9 +1,51 @@
 import math
-
 import torch
 
+from colossalai.registry import OPTIMIZERS
+
 
+@OPTIMIZERS.register_module
 class CPUAdam(torch.optim.Optimizer):
+    """Implements Adam algorithm.
+
+    Supports both GPU and CPU. Requires ColossalAI to be installed via
+    ``pip install .``.
+
+    This version of CPU Adam accelates parameters updating on CPU with SIMD.
+    Support of AVX2 or AVX512 is required.
+
+    The GPU part is an naive implementation.
+
+    CPU Adam also supports the hybrid calculation of fp16 and fp32.
```

Comment:
support the (what tensor is int format of) fp16 and fp32?

Reply:
I have updated the code and declared it.